packages/interaction-kit/src/interactions/autocomplete/autocomplete-interaction.ts

```diff
@@ -0,0 +1,83 @@
+import Application from "../../application";
+import { Autocomplete, RequestBody, ResponseHandler } from "../../interfaces";
+import { AutocompleteInteractionResponseTypes } from "./types";
+import {
+	APIApplicationCommandAutocompleteInteraction,
+	APIApplicationCommandAutocompleteResponse,
+	APIApplicationCommandInteractionDataOption,
+	APIApplicationCommandOptionChoice,
+	APIInteractionGuildMember,
+} from "discord-api-types/v9";
+import { ResponseStatus } from "../../requests/response";
+import type { Snowflake } from "../../structures/snowflake";
+import { Choices } from "../../commands/options/choices";
+import {
+	APIApplicationCommandInteractionDataAutocompleteOption,
+	isAutocompleteOption,
+} from "../../commands/options/option";
+
+// TODO: Specify string or number?
+export default class AutocompleteInteraction<
+	T extends APIApplicationCommandOptionChoice
+> implements Autocomplete<T>
+{
+	public readonly name: string;
+	public readonly token: string;
+	// TODO: Should this be called "focused"?
+	public readonly target: APIApplicationCommandInteractionDataAutocompleteOption;
+
+	public readonly respond: ResponseHandler<APIApplicationCommandAutocompleteResponse>;
+	public readonly options: Map<
+		string,
+		APIApplicationCommandInteractionDataOption
+	> = new Map();
+
+	// TODO: Convert these into Records
+	public readonly channelID: Snowflake | undefined;
+	public readonly guildID: Snowflake | undefined;
```

Comment:
Are these going to be consistent with the other changes to `Id`?

Reply:
It's already changed [here](https://github.com/IanMitchell/interaction-kit/pull/114/files#diff-69444ade4e63c8e717afbb4aeac37719227e2d48db6ffdfba2ff19fa2f89ffa5L36-R37).